### PR TITLE
CLDC-2490 Tidy up DSA model

### PR DESCRIPTION
**Pull request by ghost**
- Remove new_data_protection_confirmation? flag
- remove unique index on dpo_id / org on DPC
- add new columns to persist org and DPO data on DPC
- update import to persist org and DPO data on DPC
- update org controller to persist org and DPO data when signing DPC
- add migration to persist org and DPO data on DPC

Next steps:
- run migration on staging and prod
- further code cleanup

### CODE REVIEW

File: app/controllers/organisations_controller.rb [hunk 1]
```diff
@@ -159,28 +159,36 @@ def merge_request
...
     return render_not_found unless current_user.is_dpo?
     return render_not_found if @organisation.data_protection_confirmed?
 
     if @organisation.data_protection_confirmation
       @organisation.data_protection_confirmation.update!(
         confirmed: true,
         data_protection_officer: current_user,
-        # When it was signed
         created_at: Time.zone.now,
+        signed_at_at: Time.zone.now,
```
Comment (kosiakkatrina) [inline app/controllers/organisations_controller.rb]: typo? `singed_at_at`

File: app/controllers/organisations_controller.rb [hunk 2]
```diff
@@ -159,28 +159,36 @@ def merge_request
...
-    return render_not_found unless FeatureToggle.new_data_protection_confirmation?
     return render_not_found unless current_user.is_dpo?
     return render_not_found if @organisation.data_protection_confirmed?
 
     if @organisation.data_protection_confirmation
       @organisation.data_protection_confirmation.update!(
         confirmed: true,
         data_protection_officer: current_user,
-        # When it was signed
         created_at: Time.zone.now,
```
Comment (kosiakkatrina) [inline app/controllers/organisations_controller.rb:173]: Do we want to be updating created_at on update?

File: app/helpers/data_sharing_agreement_helper.rb [hunk 3]
```diff
@@ -61,7 +61,11 @@ def present_section_12_2(data_protection_confirmation:, user:, organisation:)
       end
     end
 
-    "12.2. For #{@org_name}: Name: #{@dpo_name}, Postal Address: #{@org_address}, E-mail address: #{@dpo_email}, Telephone number: #{@org_phone}"
+    if data_protection_confirmation&.confirmed? && @dpo_email.exclude?("@") # Do not show invalid email addresses
+      "12.2. For #{@org_name}: Name: #{@dpo_name}, Postal Address: #{@org_address}, Telephone number: #{@org_phone}"
+    else
+      "12.2. For #{@org_name}: Name: #{@dpo_name}, Postal Address: #{@org_address}, E-mail address: #{@dpo_email}, Telephone number: #{@org_phone}"
+    end
```
Comment (kosiakkatrina) [inline app/helpers/data_sharing_agreement_helper.rb:68]: I'd flip this and use `.include?` instead, but that's just preference
Would it drop into `else` if `data_protection_confirmation` is not confirmed?

File: app/models/organisation.rb [hunk 4]
```diff
@@ -112,13 +112,7 @@ def display_organisation_attributes
       { name: "Registration number", value: housing_registration_no || "", editable: false },
       { name: "Rent periods", value: rent_period_labels, editable: false, format: :bullet },
       { name: "Owns housing stock", value: holds_own_stock ? "Yes" : "No", editable: false },
-    ].compact
-
-    unless FeatureToggle.new_data_protection_confirmation?
-      attrs << { name: "Data protection agreement", value: data_protection_agreement_string, editable: false }
```
Comment (kosiakkatrina) [inline app/models/organisation.rb:118]: Do we not want to display this at all?

File: app/controllers/organisations_controller.rb (hunk 2 above)
Comment (ghost) [reply app/controllers/organisations_controller.rb:173]: I will remove this after having run the migration so keep things in sync

File: app/models/organisation.rb (hunk 4 above)
Comment (ghost) [reply app/models/organisation.rb:118]: this is the old way to display it, now this is done with a helper

File: app/controllers/organisations_controller.rb (hunk 1 above)
Comment (ghost) [reply app/controllers/organisations_controller.rb]: yes! will fix

File: app/helpers/data_sharing_agreement_helper.rb (hunk 3 above)
Comment (ghost) [reply app/helpers/data_sharing_agreement_helper.rb:68]: I think then i'd have to invert `data_protection_confirmation&.confirmed?` and that'll look as bad or worse than `.exclude`

Comment (kosiakkatrina) [reply app/helpers/data_sharing_agreement_helper.rb:68]: Sure!

**Review by kosiakkatrina — APPROVED** [review]

**Pull request merged**